gentest/gentest-driver.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+import * as fs from 'node:fs/promises';
+import {format} from 'node:util';
+import * as path from 'path';
+import * as process from 'node:process';
+import {Builder, logging} from 'selenium-webdriver';
+import {Options} from 'selenium-webdriver/chrome.js';
+
+await (async () => {
+  const options = new Options();
+  options.addArguments(
+    '--force-device-scale-factor=1',
+    '--window-position=0,0',
+    '--hide-scrollbars',
+  );
+  options.setLoggingPrefs({
+    browser: 'ALL',
+    performance: 'ALL',
+  });
+  const driver = await new Builder()
+    .forBrowser('chrome')
+    .setChromeOptions(options)
+    .build();
+
+  process.chdir(path.dirname(process.argv[1]));
+
+  const fixtures = await fs.readdir('./fixtures');
+  for (const fileName of fixtures) {
```

Comment:
It would be nice if the behaviour when the generation script (script run within the browser tab) throws an error could be improved. Specifically, it would be nice if no tests were updated in that case (and ideally that the error, along with the file/fixture that caused it, was printed to the console).

Reply:
A followup me an @NickGerleman were thinking about was to make it so that there is minimum logic being run on the browser. Right now that is where all the complicated code generation comes from because that is where we inspect the browsers layout. Ideally we port that over to being driven off of the browser if our driver library has such capabilities (not very familiar with all that you can do). Right now, because all of that logic is on the browser and we just read the console.logs, it is hard to do anything elegant around error handling and debugging.